apps/aecore/lib/aecore/txs/pool/worker.ex

```diff
@@ -126,11 +123,38 @@ defmodule Aecore.Txs.Pool.Worker do
     end
   end
 
-  @spec get_tx_size_bytes(SignedTx.t()) :: integer()
+  @spec get_tx_size_bytes(SignedTx.t()) :: non_neg_integer()
   def get_tx_size_bytes(tx) do
     tx |> :erlang.term_to_binary() |> :erlang.byte_size()
   end
 
+  @spec is_minimum_fee_met?(SignedTx.t(), :miner | :pool | :validation, non_neg_integer()) ::
+          boolean()
+  def is_minimum_fee_met?(tx, identifier, block_height \\ nil) do
+    case tx.data.payload do
+      %SpendTx{} ->
+        SpendTx.is_minimum_fee_met?(tx, identifier)
```

Comment:
If almost every child tx of DataTx is going to have the `is_minimum_fee_met?` why not have it as a callback function in the Transaction Behaviour and have a generic call to it. They this case would be one liner eg. `tx.data.payload.type.is_minimum_fee_met?(some generic arguments)`. Maybe the args could be: SpendTx and block_height, atm that will cover all the funcs.

Reply:
I think the cases differ too much for this to be useful, the SpendTx takes an identifier and none of the others besides the response need it, the SpendTx doesn't require block height and so on

Reply:
I would still consider adjusting to add `is_minimum_fee_met?` to Transaction Behaviour. It would make Oracle transactions more self contained.